### ISSUE TITLE
Import via local folder failed

### ISSUE THREAD

**Issue by hzulla** [electerious/Lychee#371]
Hi,

I tried importing ca. 250 photo files from a local directory on the server. Access rights were correct. Only 10% of the files were imported. I found no error in the logs telling me what went wrong. How to debug this?

Thanks!

**Comment by LudovicRousseau** [comment]
Have you tried to increase the amount of time allocated to the PHP scripts?

I have a Lychee patch that does exactly that. 200 seconds is not enough for me.
```
--- a/php/api.php
+++ b/php/api.php
@@ -5,7 +5,7 @@
 # @copyright   2015 by Tobias Reich
 ###
 
-@ini_set('max_execution_time', '200');
+@ini_set('max_execution_time', '3600');
 @ini_set('post_max_size', '200M');
 @ini_set('upload_max_size', '200M');
 @ini_set('upload_max_filesize', '20M');
```

**Comment by hzulla** [comment]
Thanks, I'll try that.

However, I have 12 years of photos on my disk. These were previously managed with f-spot, which now appears to be abandoned software. What is the recommend way to import a (really really) large amount of photos to Lychee?

**Comment by electerious** [comment]
12 years of photos is a lot. I would suggest to import them in smaller pieces, even if it takes some time. PHP isn't good at handling such a huge amount.

**Comment by LudovicRousseau** [comment]
@hzulla maybe you should use lycheesync (untested by me) https://github.com/GustavePate/lycheesync

**Comment by hzulla** [comment]
@LudovicRousseau thanks, lycheesync seems to be the solution I was looking for.

However, it is still puzzling that there was no error message or anything helpful for the user when Lychee's own folder import failed.

**Comment by electerious** [comment]
The current import isn't very good at handling errors. I'm currently rewriting some parts to make it better. Stay tuned.

**Comment by deidyomega** [comment]
Just as a note, if you keep clicking 'import', slowly but surely all the photos will be processed.  For my server, it would do them in chunks of about 2k per import attempt.

How I got it to work:
I used phpmyadmin to see how many records where in the system.  Once the count stopped going up, I refreshed the lyncee page, and clicked local import again.

This works because once the program processes an image, it deletes it from the upload directory.  It took me about 5 hours, but I got them all complete!

A python/perl script may be better at processing large sets of photos.
